assets/js/admin/woocommerce_admin.js

```diff
@@ -61,39 +61,53 @@
 			})
 
 			.on( 'change', '.wc_input_price[type=text], .wc_input_decimal[type=text], .wc-order-totals #refund_amount[type=text]', function() {
-				var regex;
+				var regex, decimalRegex, decimailPoint;
 
 				if ( $( this ).is( '.wc_input_price' ) || $( this ).is( '#refund_amount' ) ) {
-					regex = new RegExp( '[^\-0-9\%\\' + woocommerce_admin.mon_decimal_point + ']+', 'gi' );
+					decimailPoint = woocommerce_admin.mon_decimal_point;
+					regex = new RegExp( '[^\-0-9\%\\' + decimailPoint + ']+', 'gi' );
+					decimalRegex = new RegExp( '\\' + decimailPoint + '+', 'gi' );
 				} else {
-					regex = new RegExp( '[^\-0-9\%\\' + woocommerce_admin.decimal_point + ']+', 'gi' );
+					decimailPoint = woocommerce_admin.decimal_point;
+					regex = new RegExp( '[^\-0-9\%\\' + decimailPoint + ']+', 'gi' );
+					decimalRegex = new RegExp( '\\' + decimailPoint + '+', 'gi' );
```

Comment:
These last 2 lines are the same as in the previous branch, I guess they could be extracted out to prevent duplication, what do you think?

Reply:
I just updated the code.